### ISSUE TITLE
weird errors if unsupported column type in resultset

### ISSUE THREAD

**Issue by bjouhier** [oracle/node-oracledb#153]
I'm getting some weird errors when I run some of my queries with `resultSet: true`.

My repro program:

``` javascript
"use strict";
var oracledb = require('oracledb');

var conn = oracledb.getConnection({
	user: 'xxx',
	password: 'yyy',
	connectString: 'localhost/ORCL',
}, _);

try {
	conn.execute('DROP TABLE T1', _);
} catch (ex) {}
conn.execute('CREATE TABLE T1 (C1 NUMBER, C2 RAW(16))', _);
conn.execute("INSERT INTO T1 (C1, C2) VALUES (5, HEXTORAW('0123456789ABCDEF0123456789ABCDEF'))", _);

var sqls = [
	'Select C1 From T1', 
	'Select C1, RAWTOHEX(ROWID) From T1',
	'Select C1, RAWTOHEX(C2) From T1',
	'Select ROWID, C1, C2 From T1',
	'Select ROWID, C1 From T1',
	'Select C2, C1 From T1',
	'Select C1, C2 From T1',
];

for (var i = 0; i < sqls.length; i++) {
	console.error("SQL=" + sqls[i]);
	try {
		var result = conn.execute(sqls[i], [], { resultSet: true }, _);
		var row = result.resultSet.getRow(_);
		console.error("ROW=" + JSON.stringify(row));	
	} catch (ex) {
		console.error(ex.message);
	}
}
```
To run it:
``` sh
npm install -g streamline
<save file as bug._js, not bug.js>
_node bug
```

Output on OSX node 0.10.35:
```
SQL=Select C1 From T1
ROW=[5]
SQL=Select C1, RAWTOHEX(ROWID) From T1
ROW=[5,"00028FF8010121970000"]
SQL=Select C1, RAWTOHEX(C2) From T1
ROW=[5,"0123456789ABCDEF0123456789ABCDEF"]
SQL=Select ROWID, C1, C2 From T1
ORA-24374: define not done before fetch or execute and fetch
SQL=Select ROWID, C1 From T1
ORA-24374: define not done before fetch or execute and fetch
SQL=Select C2, C1 From T1
ORA-24374: define not done before fetch or execute and fetch
SQL=Select C1, C2 From T1
Segmentation fault: 11
```

The `ORA-24374` error is confusing. Also, I get this strange error if the column with unsupported data type comes first (`Select C2, C1`), but I get a segfault if it comes second (`Select C1, C2`).

With `resultSet: false`, I get a much better error message: `NJS-010: unsupported data type in select list` and I don't get any segfault. 

Of course, we're anxious to get LOBS support, with the ability to get small RAW values as `Buffers` and large BLOBS as binary streams. But in the mean time, it would be good to improve handling of unsupported types when `resultSet` is true.

**Comment by cjbj** [comment]
This is a buglet. Thanks for the report. Unsupported types should give NJS-010 for both result sets and non result sets.  We'll look into it.

**Comment by cjbj** [comment]
@bjouhier this should be fixed in node-oracledb 1.0
